tests/test_edgeql_functions.py

```diff
@@ -1239,3 +1243,1141 @@ class TestEdgeQLFunctions(tb.QueryTestCase):
+            SELECT math::stddev({0.1, 0.1, -0.1, 0.1});
+            SELECT math::stddev(<decimal>{0.1, 0.2, 0.3});
+        ''', [
+            {0},
+            {1.0},
+            {1.0},
+            {0.1},
+            {0.1},
+        ])
+
+    async def test_edgeql_functions_math_stddev_02(self):
+        await self.assert_query_result(r'''
+            SELECT math::stddev(<int16>{1, 1}) IS decimal;
+            SELECT math::stddev(<int32>{1, 1}) IS decimal;
+            SELECT math::stddev(<int64>{1, 1}) IS decimal;
+            SELECT math::stddev(<float32>{1, 1}) IS float64;
+            SELECT math::stddev(<float64>{1, 1}) IS float64;
+            SELECT math::stddev(<decimal>{1, 1}) IS decimal;
+        ''', [
+            {True},
+            {True},
+            {True},
+            {True},
+            {True},
+            {True},
+        ])
+
+    # BUG: we don't yet validate that the return cardinality is 1 here
+    #
+    # Standard deviation is not defined for sets with fewer than 2 elements.
+    @unittest.expectedFailure
+    async def test_edgeql_functions_math_stddev_03(self):
+        with self.assertRaisesRegex(
+                exc.UnknownEdgeDBError,
+                r"stddev in undefined for input set.+< 2"):
+            await self.con.execute(r'''
+                SELECT math::stddev(<int64>{});
+            ''')
+
+    @unittest.expectedFailure
+    async def test_edgeql_functions_math_stddev_04(self):
+        with self.assertRaisesRegex(
+                exc.UnknownEdgeDBError,
+                r"stddev in undefined for input set.+< 2"):
+            await self.con.execute(r'''
+                SELECT math::stddev(1);
+            ''')
+
+    async def test_edgeql_functions_math_stddev_pop_01(self):
+        await self.assert_query_result(r'''
+            SELECT math::stddev_pop(1);
+            SELECT math::stddev_pop({1, 1, 1});
+            SELECT math::stddev_pop({1, 2, 1, 2});
+            SELECT math::stddev_pop({0.1, 0.1, 0.1});
+            SELECT math::stddev_pop({0.1, 0.2, 0.1, 0.2});
+        ''', [
+            {0},
+            {0},
+            {0.5},
+            {0},
+            {0.05},
+        ])
+
+    async def test_edgeql_functions_math_stddev_pop_02(self):
+        await self.assert_query_result(r'''
+            SELECT math::stddev_pop(<int16>1) IS decimal;
+            SELECT math::stddev_pop(<int32>1) IS decimal;
+            SELECT math::stddev_pop(<int64>1) IS decimal;
+            SELECT math::stddev_pop(<float32>1) IS float64;
+            SELECT math::stddev_pop(<float64>1) IS float64;
+            SELECT math::stddev_pop(<decimal>1) IS decimal;
+        ''', [
+            {True},
+            {True},
+            {True},
+            {True},
+            {True},
+            {True},
+        ])
+
+    # BUG: we don't yet validate that the return cardinality is 1 here
+    #
+    # Population standard deviation is not defined for an empty set.
+    @unittest.expectedFailure
```

Comment:
Please rebase onto the most recent `master` branch and start using `@edb.tools.test.not_implemented(reason)` and `@edb.tools.test.xfail(reason)` decorators.

Reply:
I've rebased and updated the tests.